lib/cirro_io_v2/responses/base.rb

```diff
@@ -0,0 +1,33 @@
+module CirroIOV2
+  module Responses
+    module Base
+      def initialize(body)
+        body = body.deep_symbolize_keys
+        if body[:object] == 'list'
+
+          list_item_class = self.class.name.gsub('List', '').constantize
+          values_hash = body.slice(*members.excluding(:data)).merge(
+            data: body[:data].map { |list_item| list_item_class.new(list_item) }
+          )
+          super(*self.members.map { |attr| values_hash[attr] })
+        elsif has_list?(body)
+          super(
+            *body.slice(*members.excluding(has_list?(body))).values,
+            create_sub_resource(has_list?(body), body[has_list?(body)])
+          )
+        else
+          super(*body.slice(*members).values)
```

Comment:
what is `members` ?

Reply:
It is a list of attributes of Struct. For example: struct = Struct.new(:foo, :bar). struct.members => [:foo, :bar]